### PR TITLE
Provide a `get_value_into` function to return a primitive type from `Feature` and `Property`

### PR DESCRIPTION
**Pull request by jgsogo**
* Adds a new `get_value_into` method to `Feature` and `Property`, the user can return the underlying type directly (see https://github.com/IBM/appconfiguration-rust-sdk/pull/30#issuecomment-2545096111):

    ```rust
    let feature = client.get_feature("my_f64_feature")?;
    let value: f64 = feature.get_value_into(entity)?;
    ```

 * Document `Feature` and `Property` traits adding usage example to the docs (these examples are compilation tests for these new methods). (#6 )

### CODE REVIEW

File: src/client/feature_proxy.rs [hunk 1]
```diff
@@ -49,6 +49,10 @@ impl<'a> Feature for FeatureProxy<'a> {
     fn get_value(&self, entity: &impl Entity) -> crate::errors::Result<Value> {
         self.client.get_feature(&self.feature_id)?.get_value(entity)
     }
+
+    fn get_value_t<T: TryFrom<Value, Error = crate::Error>>(&self, entity: &impl Entity) -> crate::errors::Result<T> {
```
Comment (atilag) [inline src/client/feature_proxy.rs:53]: `TryFrom<>`!! brilliant :) 
I was very unhappy with the boilerplate that was driving me the `Form<> + Form<> ...` 

I can imagine where the `_t` convention is coming from, looked very familiar to my C++ mindset :)
But I would definitely call it:
```
get_value_by_type()
```
and make it even more explicit. Does it make sense?

**Review by atilag — COMMENTED** [review]
I like how the interface looks like for the user.
I would write some tests, though.

**Comment by jgsogo** [comment]
> I would write some tests, though.

@atilag , there are tests, documentation tests 😉

**Comment by jgsogo** [comment]
`get_value_into`?

**Review by atilag — APPROVED** [review]

**Pull request merged**